processor/swok8sworkloadtypeprocessor/processor.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2025 SolarWinds Worldwide, LLC. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package swok8sworkloadtypeprocessor // import "github.com/solarwinds/solarwinds-otel-collector-releases/processor/swok8sworkloadtypeprocessor"
+import (
+	"context"
+	"fmt"
+	"iter"
+	"maps"
+	"slices"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/processor"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+type swok8sworkloadtypeProcessor struct {
+	logger    *zap.Logger
+	cancel    context.CancelFunc
+	config    *Config
+	settings  processor.Settings
+	k8sClient kubernetes.Interface
+	factory   informers.SharedInformerFactory
+	informers map[string]cache.SharedIndexInformer
+}
+
+func (cp *swok8sworkloadtypeProcessor) getAttribute(attributes pcommon.Map, attrName string) string {
+	if attrVal, ok := attributes.Get(attrName); ok && attrVal.Type() == pcommon.ValueTypeStr {
+		return attrVal.Str()
+	}
+	return ""
+}
+
+type dataPointSlice[DP dataPoint] interface{ All() iter.Seq2[int, DP] }
+type dataPoint interface{ Attributes() pcommon.Map }
+
+func processDatapoints[DPS dataPointSlice[DP], DP dataPoint](cp *swok8sworkloadtypeProcessor, datapoints DPS) {
+	for _, dp := range datapoints.All() {
+		attributes := dp.Attributes()
+		for _, workloadMapping := range cp.config.WorkloadMappings {
+			name := cp.getAttribute(attributes, workloadMapping.NameAttr)
+			if name == "" {
+				continue
+			}
+			namespace := cp.getAttribute(attributes, workloadMapping.NamespaceAttr)
+			var workloadKey string
+			if namespace != "" {
+				workloadKey = fmt.Sprintf("%s/%s", namespace, name)
+			} else {
+				workloadKey = name
+			}
+
+			for _, workloadType := range workloadMapping.ExpectedTypes {
+				workload, exists, err := cp.informers[workloadType].GetStore().GetByKey(workloadKey)
+				if err != nil {
+					cp.logger.Error("error getting workload from cache", zap.String("workloadType", workloadType), zap.String("workloadKey", workloadKey), zap.Error(err))
```

Comment:
nit: This situation can happen - when entity get deleted and cacheStore gets updated before telemetry about communication gets processed, so I am not sure if I would log it as error, warn would be probably more appropriate.